internal/profiling/server.go

```diff
@@ -0,0 +1,11 @@
+package profiling
+
+import (
+	"fmt"
+	"net/http"
+	_ "net/http/pprof"
```

Comment:
please use a concrete import here, not the side effects based one, as this will automatically start pprof on the default handler (https://go.dev/src/net/http/pprof/pprof.go look into the init function). if you want to do it manually:

https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/manager/internal.go#L312-L327 see here for an example